platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventValidationChecks.java

```diff
@@ -0,0 +1,268 @@
+            @NonNull final RateLimitedLogger logger) {
+
+        final boolean generationValid = event.getGeneration() >= minimumGenerationNonAncient;
+        if (!generationValid) {
+            logger.error(
+                    INVALID_EVENT_ERROR.getMarker(),
+                    "Event generation is invalid. Event: {}, Minimum Generation non ancient: {}",
+                    event,
+                    minimumGenerationNonAncient);
+        }
+
+        return generationValid;
+    }
+
+    /**
+     * Determine whether a given event has a valid creation time.
+     *
+     * @param event             the event to be validated
+     * @param logger            a logger for validation errors
+     * @param metricAccumulator for counting occurrences of invalid event creation times
+     * @return true if the creation time of the event is strictly after the creation time of its self-parent, otherwise false
+     */
+    public static boolean isValidTimeCreated(
+            @NonNull final EventImpl event,
+            @NonNull final RateLimitedLogger logger,
+            @NonNull final LongAccumulator metricAccumulator) {
+        final EventImpl selfParent = event.getSelfParent();
+
+        final boolean validTimeCreated =
+                selfParent == null || event.getTimeCreated().isAfter(selfParent.getTimeCreated());
+
+        if (!validTimeCreated) {
+            logger.error(
+                    INVALID_EVENT_ERROR.getMarker(),
+                    "Event timeCreated is invalid. Event: {}, Time created: {}, Parent created: {}",
+                    event.toMediumString(),
+                    event.getTimeCreated(),
+                    selfParent.getTimeCreated());
+            metricAccumulator.update(1);
+        }
+
+        return validTimeCreated;
+    }
+
+    /**
+     * Determine whether a given event has valid parents.
+     *
+     * @param event             the event to be validated
+     * @param singleNodeNetwork true if the network is a single node network, otherwise false
+     * @param logger            a logger for validation errors
+     * @param metricAccumulator for counting occurrences of invalid event parents
+     * @return true if the event has valid parents, otherwise false
+     */
+    public static boolean areParentsValid(
+            @NonNull final EventImpl event,
+            final boolean singleNodeNetwork,
+            @NonNull final RateLimitedLogger logger,
+            @NonNull final LongAccumulator metricAccumulator) {
+
+        final BaseEventHashedData hashedData = event.getHashedData();
+
+        final Hash selfParentHash = hashedData.getSelfParentHash();
+        final long selfParentGeneration = hashedData.getSelfParentGen();
+
+        // If a parent hash is missing, then the generation must also be invalid.
+        // If a parent hash is not missing, then the generation must be valid.
+        if (selfParentHash == null != selfParentGeneration < FIRST_GENERATION) {
```

Comment:
For the sake of readability, perhaps add parenthesis

```suggestion
        if ((selfParentHash == null) != (selfParentGeneration < FIRST_GENERATION)) {
```